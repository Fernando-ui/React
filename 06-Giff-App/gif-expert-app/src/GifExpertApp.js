import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';

const GifExpertApp = () => {

   
    const [categories, setCategories] = useState(['One puch','Samurai', 'Dragon Ballz']);

    // const handleAdd = ()=>{

    //     setCategories([...categories, 'One Piece' ]);

    // }
    return (

        <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories = { setCategories } />
        <hr/>
        
        <ol>
            {
                categories.map( ( element )=> {

                    return <li key = {element}>{element}</li>
                })
            }
        </ol>

        </>
    )

}

export default GifExpertApp;