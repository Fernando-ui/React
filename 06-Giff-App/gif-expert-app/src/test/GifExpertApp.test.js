import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('Haciendo pruebas con el componente <GifExpertApp />', () => {
    
    
    test('El snapshot debe de coincidir', () => {
        
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
        
    });
    
    test('Debe de mostrar una lista de categorias', () => {
        
        const categories = ['One punch', 'dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length  )  ;
    })
    

});
