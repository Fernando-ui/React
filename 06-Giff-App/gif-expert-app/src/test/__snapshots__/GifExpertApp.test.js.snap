// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Haciendo pruebas con el componente <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="dragon Ball"
      key="dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Haciendo pruebas con el componente <GifExpertApp /> El snapshot debe de coincidir 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
